--- xine-lib-1.2.3/m4/input.m4~	2015-03-09 19:21:30.000000000 +0100
+++ xine-lib-1.2.3/m4/input.m4	2015-03-09 19:42:52.059143679 +0100
@@ -126,9 +126,11 @@
     dnl XXX: This could be cleaned up so that code does not have to ifdef so much
     XINE_ARG_WITH([external-dvdnav], [Use external dvdnav library (not recommended)])
     if test x"$with_external_dvdnav" != x"no"; then
-        ACX_PACKAGE_CHECK([DVDNAV], [0.1.9], [dvdnav-config],
-                          [AC_DEFINE([HAVE_DVDNAV], 1, [Define this if you have a suitable version of libdvdnav])],
-                          [AC_MSG_RESULT([*** no usable version of libdvdnav found, using internal copy ***])])
+        no_dvdnav=no
+        PKG_CHECK_MODULES([DVDNAV], [dvdnav >= 0.1.9], [], [no_dvdnav=yes])
+	if test "$no_dvdnav" = 'no'; then
+            AC_DEFINE([HAVE_DVDNAV], 1, [Define this if you have a suitable version of libdvdnav])
+	fi
         AC_CHECK_LIB([dvdread], [navRead_DSI], [DVDNAV_LIBS="$DVDNAV_LIBS -ldvdread"], [])
     else
         AC_MSG_RESULT([Using included DVDNAV support])
