diff -urNp -x '*.orig' xine-lib-1.2.8.org/src/post/goom/convolve_fx.c xine-lib-1.2.8/src/post/goom/convolve_fx.c
--- xine-lib-1.2.8.org/src/post/goom/convolve_fx.c	2017-02-21 12:13:13.000000000 +0100
+++ xine-lib-1.2.8/src/post/goom/convolve_fx.c	2022-02-14 00:14:02.757106126 +0100
@@ -152,7 +152,7 @@ static void create_output_with_brightnes
     yprime += c;
 
 #if defined(HAVE_MMX) && !defined(__sun)
-#if defined(ARCH_X86_64)
+#if defined(ARCH_X86_64) && !defined(__ILP32__)
     /* 64-bit os version. */
     __asm__ __volatile__
       ("\n\t pxor  %%mm7,  %%mm7"  /* mm7 = 0   */
diff -urNp -x '*.orig' xine-lib-1.2.8.org/src/post/planar/noise.c xine-lib-1.2.8/src/post/planar/noise.c
--- xine-lib-1.2.8.org/src/post/planar/noise.c	2017-02-21 12:13:13.000000000 +0100
+++ xine-lib-1.2.8/src/post/planar/noise.c	2022-02-14 00:14:02.757106126 +0100
@@ -31,7 +31,7 @@
 #include <math.h>
 #include <pthread.h>
 
-#ifdef ARCH_X86_64
+#if defined(ARCH_X86_64) && !defined(__ILP32__)
 #  define REG_a  "rax"
 #  define intarch_t int64_t
 #else
