diff -urN xine-lib-1-rc3b.orig/src/audio_out/audio_oss_out.c xine-lib-1-rc3b/src/audio_out/audio_oss_out.c
--- xine-lib-1-rc3b.orig/src/audio_out/audio_oss_out.c	2004-04-01 19:53:05.229746016 +0200
+++ xine-lib-1-rc3b/src/audio_out/audio_oss_out.c	2004-04-01 20:18:47.877227952 +0200
@@ -57,10 +57,8 @@
 #   include <sys/soundcard.h>
 #  endif
 #else
-# if defined(__linux__)
-#  include <linux/config.h> /* Check for DEVFS */
-# endif
 # include <sys/soundcard.h>
+# include <sys/stat.h>
 #endif
 #include <sys/ioctl.h>
 #include <inttypes.h>
