--- xine-lib-1.1.1/configure.ac.orig	2005-11-19 23:50:49.494038504 +0100
+++ xine-lib-1.1.1/configure.ac	2005-11-20 00:07:16.747953096 +0100
@@ -1869,6 +1869,8 @@
 enable_ffmmx="no"
 enable_armv4l="no"
 
+arch_sparc="no"
+
 case "$host_or_hostalias" in
   i?86-* | k?-* | athlon-* | pentium*-)
     dnl enable x86 specific parts of the code
@@ -1933,6 +1935,7 @@
     fi
     ;;
   sparc*-*-linux*)
+    arch_sparc="yes"
     if test x$enable_vis = xyes; then
       AC_DEFINE_UNQUOTED(ENABLE_VIS,,[Define this if you have Sun UltraSPARC CPU])
     fi
@@ -1982,6 +1985,7 @@
 fi
 AC_SUBST(W32DLL_DEP)
 
+AM_CONDITIONAL([ARCH_SPARC], test x"$arch_sparc" = "xyes")
 AM_CONDITIONAL(HAVE_FFMMX, test x"$enable_ffmmx" = "xyes")
 
 dnl ---------------------------------------------
diff -urbB xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am
--- xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am	2004-04-25 20:59:58.000000000 +0200
+++ xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am	2005-09-16 12:31:05.607968352 +0200
@@ -13,3 +13,9 @@
 sparc_modules = $(libavcodec_sparc_src)
 
 libavcodec_sparc_la_SOURCES = $(sparc_modules) $(libavcodec_sparc_dummy)
+
+if ARCH_SPARC
+# this file needs 64 float regs (ultrasparc) and is not executed on !ultra
+dsputil_vis.lo: dsputil_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
diff -urbB xine-lib-1.1.0.org/src/libmpeg2/Makefile.am xine-lib-1.1.0/src/libmpeg2/Makefile.am
--- xine-lib-1.1.0.org/src/libmpeg2/Makefile.am	2005-04-26 00:44:22.000000000 +0200
+++ xine-lib-1.1.0/src/libmpeg2/Makefile.am	2005-09-16 12:31:36.309301040 +0200
@@ -31,3 +31,9 @@
 
 noinst_HEADERS = vlc.h mpeg2.h xvmc.h xvmc_vld.h mpeg2_internal.h idct_mlib.h vis.h \
 	libmpeg2_accel.h
+
+if ARCH_SPARC
+motion_comp_vis.lo: motion_comp_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
+	
diff -urbB xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am
--- xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am	2006-04-12 06:25:56.000000000 +0000
+++ xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am	2006-04-12 07:34:04.000000000 +0000
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.common
 
-AM_CFLAGS = $(LIBFFMPEG_CFLAGS)
+AM_CFLAGS = $(LIBFFMPEG_CFLAGS) -I$(top_srcdir)/src/libffmpeg/libavutil
 ASFLAGS =
 
 noinst_LTLIBRARIES = libavcodec_sparc.la
