--- xine-lib-1.2.6/src/libw32dll/common.c.orig	2015-10-07 18:08:58.268849455 +0200
+++ xine-lib-1.2.6/src/libw32dll/common.c	2015-10-07 18:13:02.622172569 +0200
@@ -6,9 +6,8 @@
   DIR                *dir;
   const char         *path, *cfgpath;
   const char * const listpath[] = { "",
-                                     "/usr/lib/codecs",
+                                     WIN32_PATH,
                                      "/usr/local/lib/codecs",
-                                     "/usr/lib/win32",
                                      "/usr/local/lib/win32",
                                      NULL };
   int                 i = 0;
