diff -urbB xine-lib-1.1.0.org/configure.ac xine-lib-1.1.0/configure.ac
--- xine-lib-1.1.0.org/configure.ac	2005-07-21 23:10:10.000000000 +0200
+++ xine-lib-1.1.0/configure.ac	2005-09-16 13:24:01.605143392 +0200
@@ -1749,6 +1749,8 @@
 enable_ffmmx="no"
 enable_armv4l="no"
 
+arch_sparc="no"
+
 case "$host_or_hostalias" in
   i386-*-freebsd*)
     CFLAGS="-pipe -fomit-frame-pointer $f_af $f_al $f_aj $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math -funroll-loops $INLINE_FUNCTIONS $CFLAGS"
@@ -1942,6 +1944,7 @@
     ;;
 
   sparc*-*-linux*)
+    arch_sparc="yes"
     case $host_alias in
     sparc-*)    cpu_cflags="-mcpu=supersparc -mtune=supersparc" ;;
     sparc64-*)
@@ -2092,6 +2095,7 @@
 fi
 AC_SUBST(W32DLL_DEP)
 
+AM_CONDITIONAL([ARCH_SPARC], test x"$arch_sparc" = "xyes")
 AM_CONDITIONAL(HAVE_FFMMX, test x"$enable_ffmmx" = "xyes")
 
 dnl ---------------------------------------------
Tylko w xine-lib-1.1.0: configure.ac~
diff -urbB xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am
--- xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am	2004-04-25 20:59:58.000000000 +0200
+++ xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am	2005-09-16 12:31:05.607968352 +0200
@@ -13,3 +13,9 @@
 sparc_modules = $(libavcodec_sparc_src)
 
 libavcodec_sparc_la_SOURCES = $(sparc_modules) $(libavcodec_sparc_dummy)
+
+if ARCH_SPARC
+# this file needs 64 float regs (ultrasparc) and is not executed on !ultra
+dsputil_vis.lo: dsputil_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
diff -urbB xine-lib-1.1.0.org/src/libmpeg2/Makefile.am xine-lib-1.1.0/src/libmpeg2/Makefile.am
--- xine-lib-1.1.0.org/src/libmpeg2/Makefile.am	2005-04-26 00:44:22.000000000 +0200
+++ xine-lib-1.1.0/src/libmpeg2/Makefile.am	2005-09-16 12:31:36.309301040 +0200
@@ -31,3 +31,9 @@
 
 noinst_HEADERS = vlc.h mpeg2.h xvmc.h xvmc_vld.h mpeg2_internal.h idct_mlib.h vis.h \
 	libmpeg2_accel.h
+
+if ARCH_SPARC
+motion_comp_vis.lo: motion_comp_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
+	
