--- xine-lib-1-beta4/src/libw32dll/wine/Makefile.am.orig	Tue Dec 24 17:05:05 2002
+++ xine-lib-1-beta4/src/libw32dll/wine/Makefile.am	Wed Feb  5 22:17:52 2003
@@ -73,10 +73,10 @@
 	wrapper.h
 
 stubs.lo: stubs.s
-	$(CC) -c $(srcdir)/stubs.s -o stubs.lo
+	$(LIBTOOL) $(CC) --mode=compile -c $(srcdir)/stubs.s -o stubs.lo
 
 wrapper.lo: wrapper.S
-	$(CC) -c $(srcdir)/wrapper.S -o wrapper.lo
+	$(LIBTOOL) $(CC) --mode=compile -c $(srcdir)/wrapper.S -o wrapper.lo
 
 debug:
 	@$(MAKE) CFLAGS="$(DEBUG_CFLAGS)"
