--- xine-lib-1-rc1/configure.ac.orig	2003-10-05 15:08:36.000000000 +0200
+++ xine-lib-1-rc1/configure.ac	2003-11-16 21:03:53.000000000 +0100
@@ -326,7 +326,6 @@
 dnl Check for OpenGL & [GLut | GLU]
 dnl ---------------------------------------------
 AC_CHECK_LIB(GL, glBegin,
-	[AC_CHECK_HEADER(GL/gl.h,
              [ ac_have_opengl="yes"
                OPENGL_LIBS="-lGL" 
 	       AC_CHECK_LIB(glut, glutInit, dnl check for glut
@@ -344,9 +343,8 @@
                                            [$X_LIBS $X_PRE_LIBS $OPENGL_LIBS -lGLU -lm $X_EXTRA_LIBS])], 
 	                    [$X_LIBS $X_PRE_LIBS -lglut $X_EXTRA_LIBS])
              ],
-	[])],
-   [],
-   [$X_LIBS $X_PRE_LIBS -lGL $X_EXTRA_LIBS])
+	     [],
+             [$X_LIBS $X_PRE_LIBS -lGL $X_EXTRA_LIBS])
 AC_SUBST(OPENGL_LIBS)
 AC_SUBST(GLUT_LIBS)
 AC_SUBST(GLU_LIBS)
@@ -414,6 +412,7 @@
 AC_ARG_WITH(xv-path,[  --with-xv-path=path     Where $xv_lib is installed],
             xv_path="$withval", xv_path="/usr/X11R6/lib")
 
+xv_lib="libXv.so"
 AC_CHECK_LIB(Xv, XvShmCreateImage, 
              [ AC_MSG_CHECKING(for $xv_lib location)
                if test -f "$xv_path/$xv_lib"; then
@@ -421,9 +420,6 @@
                  XV_LIB="-L$xv_path -lXv"
 	         AC_DEFINE(HAVE_XV,1,[Define this if you have libXv installed])
 	         ac_have_xv="yes"
-		 if test x$xv_lib = "xlibXv.a" ; then
-		   AC_DEFINE(HAVE_XV_STATIC,1,[Define this if you have libXv.a])
-		 fi
                else
                  AC_MSG_RESULT(not found in $xv_path)
 	         echo
