diff -urN xine-lib-1-beta4.orig/m4/Makefile.am xine-lib-1-beta4/m4/Makefile.am
--- xine-lib-1-beta4.orig/m4/Makefile.am	Thu Jan 30 14:28:10 2003
+++ xine-lib-1-beta4/m4/Makefile.am	Thu Jan 30 14:28:45 2003
@@ -2,11 +2,7 @@
 ## Process this file with automake to produce Makefile.in
 ##
 
-if INSTALL_M4
 m4datadir = @ACLOCAL_DIR@
-else
-m4datadir = $(datadir)/aclocal
-endif
 
 m4data_DATA = xine.m4
 
