--- xine-lib-1.1.17/src/libw32dll/common.c.orig	2009-11-30 21:55:55.000000000 +0100
+++ xine-lib-1.1.17/src/libw32dll/common.c	2010-01-21 09:21:44.000000000 +0100
@@ -6,9 +6,8 @@
   DIR                *dir;
   char               *path, *cfgpath;
   char               *listpath[] = { "",
-                                     "/usr/lib/codecs",
+                                     WIN32_PATH,
                                      "/usr/local/lib/codecs",
-                                     "/usr/lib/win32",
                                      "/usr/local/lib/win32",
                                      NULL };
   int                 i = 0;
