--- xine-lib-1-rc4a/src/libffmpeg/libavcodec/sparc/Makefile.am.orig	2004-04-25 18:59:58.000000000 +0000
+++ xine-lib-1-rc4a/src/libffmpeg/libavcodec/sparc/Makefile.am	2004-06-07 20:33:46.000000000 +0000
@@ -13,3 +13,7 @@
 sparc_modules = $(libavcodec_sparc_src)
 
 libavcodec_sparc_la_SOURCES = $(sparc_modules) $(libavcodec_sparc_dummy)
+
+# this file needs 64 float regs (ultrasparc) and is not executed on !ultra
+dsputil_vis.lo: dsputil_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
--- xine-lib-1-rc4a/src/libmpeg2/Makefile.am.orig	2004-04-29 12:47:39.000000000 +0000
+++ xine-lib-1-rc4a/src/libmpeg2/Makefile.am	2004-06-07 21:31:14.000000000 +0000
@@ -29,3 +29,6 @@
 xineplug_decode_mpeg2_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
 noinst_HEADERS = vlc.h mpeg2.h xxmc.h mpeg2_internal.h idct_mlib.h vis.h
+
+motion_comp_vis.lo: motion_comp_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
