--- xine-lib-0.9.9/src/libw32dll/Makefile.am~	Thu May  2 20:19:09 2002
+++ xine-lib-0.9.9/src/libw32dll/Makefile.am	Thu May  2 20:19:09 2002
@@ -1,6 +1,6 @@
 CFLAGS = @CFLAGS@ -I$(srcdir)/wine
 
-DEBUG_CFLAGS = @DEBUG_CFLAGS@ -I$(srcdir)/wine
+INCLUDES = -I$(srcdir)/wine -I../xine-utils -I../xine-engine -I../video_out -I../audio_out -I..
 
 SUBDIRS = wine DirectShow
 
@@ -26,15 +26,6 @@
 
 noinst_HEADERS = w32codec.h libwin32.h
 
-debug:
-	@list='$(SUBDIRS)'; for subdir in $$list; do \
-	  (cd $$subdir && $(MAKE) $@) || exit;\
-	done;
-	@$(MAKE) CFLAGS="$(DEBUG_CFLAGS)"
-
-install-debug: debug
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
-
 mostlyclean-generic:
 	-rm -f *~ \#* .*~ .\#*
 
--- xine-lib-0.9.12/configure.in.orig	Wed Jun 26 22:50:01 2002
+++ xine-lib-0.9.12/configure.in	Wed Jun 26 22:53:40 2002
@@ -155,7 +155,7 @@
 AC_C_ALWAYS_INLINE
 AC_TYPE_OFF_T
 AC_TYPE_SIZE_T
-AM_TYPE_PTRDIFF_T
+AC_CHECK_TYPES([ptrdiff_t])
 
 dnl
 dnl debug cflags
--- xine-lib-0.9.12/src/libw32dll/wine/Makefile.am.orig	Tue Mar 26 00:07:17 2002
+++ xine-lib-0.9.12/src/libw32dll/wine/Makefile.am	Wed Jun 26 23:45:53 2002
@@ -33,7 +33,7 @@
 		 wineacm.h     winestring.h resource.h ldt_keeper.h com.h
 
 stubs.lo: stubs.s
-	$(CC) -c $(srcdir)/stubs.s -o stubs.lo
+	$(LIBTOOL) $(CC) -c $(srcdir)/stubs.s -o stubs.lo
 
 debug:
 	@$(MAKE) CFLAGS="$(DEBUG_CFLAGS)"
