--- xine-lib-1-rc3a/configure.ac.orig	2003-12-31 01:41:16.000000000 +0100
+++ xine-lib-1-rc3a/configure.ac	2003-12-31 01:48:45.137006072 +0100
@@ -142,7 +142,6 @@
 ALL_LINGUAS="fr pt_BR de sk es pl_PL cs it"
 AM_GNU_GETTEXT(use-libtool, need-ngettext)
 AC_PROG_GMSGFMT_PLURAL
-AC_CONFIG_LINKS($nls_cv_header_libgt, $nls_cv_header_intl)
 AM_CONDITIONAL(INCLUDED_INTL, test x$USE_INCLUDED_LIBINTL = "xyes")
 if test x"$USE_INCLUDED_LIBINTL" = x"yes"; then
    INTLDIR="-I\$(top_builddir)/intl -I\$(top_srcdir)/intl"
