diff -urN xine-lib-0.5.3/src/libw32dll/wine/Makefile.am xine-lib-0.5.3.new/src/libw32dll/wine/Makefile.am
--- xine-lib-0.5.3/src/libw32dll/wine/Makefile.am	Tue Jul 17 03:29:41 2001
+++ xine-lib-0.5.3.new/src/libw32dll/wine/Makefile.am	Wed Sep 12 18:20:55 2001
@@ -20,7 +20,7 @@
 libwine_la_SOURCES = afl.c       elfdll.c    module.c    pe_resource.c  \
 		     resource.c  win32.c     driver.c    ext.c          \
 		     pe_image.c  registry.c  vfl.c        
-libwine_la_LIBADD = stubs.lo
+libwine_la_LIBADD = stubs.o
 
 noinst_HEADERS = avifmt.h      elfdll.h    msacm.h     pshpack1.h   winbase.h \
 		 winnt.h       basetsd.h   heap.h      msacmdrv.h   pshpack2.h\
@@ -30,8 +30,8 @@
 		 poppack.h     vfw.h       loader.h    registry.h   win32.h   \
 		 wineacm.h     winestring.h config.h   resource.h
 
-stubs.lo: stubs.s
-	$(CC) -c $(top_srcdir)/src/libw32dll/wine/stubs.s -o stubs.lo
+stubs.o: stubs.s
+	$(CC) -c $(top_srcdir)/src/libw32dll/wine/stubs.s -o stubs.o
 
 debug:
 	@$(MAKE) CFLAGS="$(DEBUG_CFLAGS)"
