--- xine-lib-1.1.4/configure.ac.orig	2007-02-01 12:35:08.818134000 +0100
+++ xine-lib-1.1.4/configure.ac	2007-02-01 12:36:20.068134000 +0100
@@ -2194,6 +2194,8 @@
 arch_x86="no"
 enable_armv4l="no"
 
+arch_sparc="no"
+
 case "$host_or_hostalias" in
   i?86-* | k?-* | athlon-* | pentium*-)
     dnl enable x86 specific parts of the code
@@ -2247,6 +2249,7 @@
     fi
     ;;
   sparc*-*-linux*)
+    arch_sparc="yes"
     if test "x$enable_vis" != xno; then
       has_vis=yes
     fi
@@ -2294,6 +2297,7 @@
    esac
 fi
 AM_CONDITIONAL(ENABLE_VIS, test "x$has_vis" = "xyes")
+AM_CONDITIONAL([ARCH_SPARC], test x"$arch_sparc" = "xyes")
 
 if test "x$arch_x86" = "xyes"; then
    AC_DEFINE_UNQUOTED(ARCH_X86,,[Define this if you're running x86 architecture])
diff -urbB xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am
--- xine-lib-1.1.0.org/src/libffmpeg/libavcodec/sparc/Makefile.am	2004-04-25 20:59:58.000000000 +0200
+++ xine-lib-1.1.0/src/libffmpeg/libavcodec/sparc/Makefile.am	2005-09-16 12:31:05.607968352 +0200
@@ -13,3 +13,9 @@
 sparc_modules = $(libavcodec_sparc_src)
 
 libavcodec_sparc_la_SOURCES = $(sparc_modules) $(libavcodec_sparc_dummy)
+
+if ARCH_SPARC
+# this file needs 64 float regs (ultrasparc) and is not executed on !ultra
+dsputil_vis.lo: dsputil_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
diff -urbB xine-lib-1.1.0.org/src/libmpeg2/Makefile.am xine-lib-1.1.0/src/libmpeg2/Makefile.am
--- xine-lib-1.1.0.org/src/libmpeg2/Makefile.am	2005-04-26 00:44:22.000000000 +0200
+++ xine-lib-1.1.0/src/libmpeg2/Makefile.am	2005-09-16 12:31:36.309301040 +0200
@@ -31,3 +31,9 @@
 
 noinst_HEADERS = vlc.h mpeg2.h xvmc.h xvmc_vld.h mpeg2_internal.h idct_mlib.h vis.h \
 	libmpeg2_accel.h
+
+if ARCH_SPARC
+motion_comp_vis.lo: motion_comp_vis.c
+	$(LTCOMPILE) -c -o $@ $< -mcpu=ultrasparc
+endif
+	
