--- xine-lib-1.1.1/configure.ac.orig	2005-11-14 13:34:45.000000000 +0100
+++ xine-lib-1.1.1/configure.ac	2005-11-19 23:50:02.836131584 +0100
@@ -517,22 +517,6 @@
 AM_PATH_OPENGL()
 
 
-dnl ---------------------------------------------
-dnl Check for platform which supports syncfb
-dnl ---------------------------------------------
-
-case "$host_or_hostalias" in
-  *linux* )
-    AC_ARG_ENABLE([fb],
-      AC_HELP_STRING([--disable-fb], [do not build syncfb plugins]),
-      have_syncfb=$enableval, have_syncfb="yes")
-    ;;
-  *)
-    have_syncfb="no"
-    ;;
-esac
-AM_CONDITIONAL(HAVE_SYNCFB, test x"$have_syncfb" = "xyes")
-
 
 dnl ----------------------------------------------
 dnl Check for usable video-for-linux (v4l) support
@@ -614,6 +598,22 @@
 esac
 AM_CONDITIONAL(PPC_ARCH, test x$ppc_arch = "xyes")
 
+dnl ---------------------------------------------
+dnl Check for platform which supports syncfb
+dnl ---------------------------------------------
+
+case "$host_or_hostalias" in
+  *linux* )
+    AC_ARG_ENABLE([fb],
+      AC_HELP_STRING([--disable-fb], [do not build syncfb plugins]),
+      have_syncfb=$enableval, have_syncfb="yes")
+    ;;
+  *)
+    have_syncfb="no"
+    ;;
+esac
+AM_CONDITIONAL(HAVE_SYNCFB, test x"$have_syncfb" = "xyes")
+
 dnl
 dnl Check if we can enable the xxmc plugin.
 dnl
@@ -2543,7 +2543,7 @@
 if test x"$no_x" != "xyes"; then
   echo "   - XShm (X11 shared memory)"
   dnl synfb
-  if test x$ac_have_syncfb = "xyes"; then
+  if test x$have_syncfb = "xyes"; then
     echo "   - SyncFB (for Matrox G200/G400 cards)"
   fi
   dnl Xv
--- xine-lib-1-rc8/src/video_out/video_out_syncfb.c.orig	2004-12-12 23:01:28.000000000 +0100
+++ xine-lib-1-rc8/src/video_out/video_out_syncfb.c	2004-12-21 21:18:33.031892224 +0100
@@ -46,6 +46,7 @@
 #include <sys/mman.h>
 #include <math.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
@@ -566,6 +567,7 @@
 
 static void syncfb_overlay_blend(vo_driver_t* this_gen, vo_frame_t* frame_gen, vo_overlay_t* overlay)
 {
+  syncfb_driver_t* this  = (syncfb_driver_t *) this_gen;
   syncfb_frame_t* frame = (syncfb_frame_t *) frame_gen;
    
   /* alpha blend here */
