--- xine-lib-1.2.1/configure.ac~	2012-02-04 01:41:43.000000000 +0100
+++ xine-lib-1.2.1/configure.ac	2012-02-15 09:05:03.683929323 +0100
@@ -267,8 +267,8 @@
 AM_CONDITIONAL([EXTERNAL_LIBXDG_BASEDIR], [test x"$have_xdg_basedir" = xyes])
 
 dnl Test for socket and network support library
-AC_CHECK_LIB([socket], [socket],        [NET_LIBS="-lsocket $NET_LIBS"])
-AC_CHECK_LIB([nsl],    [gethostbyname], [NET_LIBS="-lnsl $NET_LIBS"])
+#AC_CHECK_LIB([socket], [socket],        [NET_LIBS="-lsocket $NET_LIBS"])
+#AC_CHECK_LIB([nsl],    [gethostbyname], [NET_LIBS="-lnsl $NET_LIBS"])
 AC_CHECK_LIB([resolv], [hstrerror],     [NET_LIBS="-lresolv $NET_LIBS"])
 AC_SUBST(NET_LIBS)
 
